src/main/java/bio/terra/grammar/DatasetNameListener.java

```diff
@@ -0,0 +1,21 @@
+package bio.terra.grammar;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DatasetNameListener extends SQLBaseListener {
```

Comment:
In addition to the visitor, I would also like to learn more about the relationship between the query and listener. It looks to me like the listener is returning the final set of dataset names?

Reply:
I saw you added this in the design document, so I am happy to close this out.